modal/mount.py

```diff
@@ -331,93 +331,108 @@ async def _put_file(file_spec: FileUploadSpec) -> api_pb2.MountFile:
         logger.debug(f"Uploaded {len(uploaded_hashes)}/{n_files} files and {total_bytes} bytes in {time.time() - t0}s")
         return _MountHandle._from_id(resp.mount_id, resolver.client, None)
 
+    @staticmethod
+    def create_client_mount():
+        import modal
 
-Mount, AioMount = synchronize_apis(_Mount)
+        # Get the base_path because it also contains `modal_utils` and `modal_proto`.
+        base_path, _ = os.path.split(modal.__path__[0])
 
+        # TODO(erikbern): this is incredibly dumb, but we only want to include packages that start with "modal"
+        # TODO(erikbern): merge functionality with _function_utils._is_modal_path
+        prefix = os.path.join(base_path, "modal")
 
-def _create_client_mount():
-    # TODO(erikbern): make this a static method on the Mount class
-    import modal
+        def condition(arg):
+            return module_mount_condition(arg) and arg.startswith(prefix)
 
-    # Get the base_path because it also contains `modal_utils` and `modal_proto`.
-    base_path, _ = os.path.split(modal.__path__[0])
+        return _Mount.from_local_dir(base_path, remote_path="/pkg/", condition=condition, recursive=True)
 
-    # TODO(erikbern): this is incredibly dumb, but we only want to include packages that start with "modal"
-    # TODO(erikbern): merge functionality with _function_utils._is_modal_path
-    prefix = os.path.join(base_path, "modal")
+    @staticmethod
+    def _get_client_mount():
+        if config["sync_entrypoint"]:
+            return _Mount.create_client_mount()
+        else:
+            return _Mount.from_name(client_mount_name(), namespace=api_pb2.DEPLOYMENT_NAMESPACE_GLOBAL)
 
-    def condition(arg):
-        return module_mount_condition(arg) and arg.startswith(prefix)
+    @staticmethod
+    def create_package_mounts(module_names: Sequence[str]) -> List["_Mount"]:
```

Comment:
should we maybe rewrite this to something that just takes a single module name? not sure if the list form is needed